### PR TITLE
Add KeepassXC Extension

### PR DESCRIPTION
**Pull request by zkytech**
## Description

Add KeepassXC Extension to access KeePass Database

> requires [KeepassXC.app](https://keepassxc.org) installed

## Type of change

- New extension

## Screencast
example

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zkytech 👋 

Cool you look into your README – I was not able to test the extension by reading it - maybe a step by step guide from scratch (Just installed [KeePassXC](https://keepassxc.org/))

**Comment by zkytech** [comment]
> `Test Guide.md` has been added.

File: extensions/keepassxc/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Keepass Extension for Raycast
```
Comment (pernielsentikaer) [inline extensions/keepassxc/README.md:1]:
```suggestion
# KeepassXC Extension for Raycast
```

File: extensions/keepassxc/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+This is a raycast extension to access keepass database.
```
Comment (pernielsentikaer) [inline extensions/keepassxc/README.md:3]:
```suggestion
This is a Raycast extension to access keepass database.
```

File: extensions/keepassxc/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- path of `KeePassXC.app` (default `/Applications/KeePassXC.app`)
+- a keepass database file
+- password to access database
+
+![preference](preferences.png)
+## Usage
+
+![example](example.gif)
+
+### Short Cuts
```
Comment (pernielsentikaer) [inline extensions/keepassxc/README.md:22]:
```suggestion
### Shortcuts
```

File: extensions/keepassxc/package.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  "title": "KeepassXC",
+  "description": "Access Keepass Database with KeepassXC-cli. Please Make Sure KeepassXC is Installed",
+  "icon": "keepassxc.png",
+  "author": "zkytech",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "KeepassXC",
+      "subtitle": "Search Password in KeepassXC",
```
Comment (pernielsentikaer) [inline extensions/keepassxc/package.json:13]:
```suggestion
      "title": "Search Passwords",
      "subtitle": "KeepassXC",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zkytech 👋 

I have some feeedback for you 🔥 

- When both app location/file location or password is wrong, all three states should show the failure toast

- Is it keepass, KeePass, KeepassXC, some more consistent naming would be nice.

- Remove command-icon.png from assets since it's redundant

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by zkytech** [comment]
> Hi @zkytech 👋
> 
> I have some feeedback for you 🔥
> 
> * When both app location/file location or password is wrong, all three states should show the failure toast
> * Is it keepass, KeePass, KeepassXC, some more consistent naming would be nice.
> * Remove command-icon.png from assets since it's redundant
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

@pernielsentikaer 

Thanks for your suggestion.

- [x] An error handler has been added to handle different error messages.
- [x] Your commits has been accepted to update words in `README.md` and `package.json`.
- [x] Redundant assets has been removed.

File: extensions/keepassxc/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import { ActionPanel, closeMainWindow, List, showToast, ToastStyle } from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/keepassxc/src/index.tsx:1]:
```suggestion
import { ActionPanel, closeMainWindow, Icon, List, showToast, ToastStyle } from "@raycast/api";
```

File: extensions/keepassxc/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                  .split("/")
+                  .slice(1, entry.split("/").length - 1)
+                  .join("\t")
+              : ""
+          }
+          keywords={entry.split("/").slice(1)}
+          actions={
+            <ActionPanel>
+              <ActionPanel.Item
+                title="Copy and Paste Password"
```
Comment (pernielsentikaer) [inline extensions/keepassxc/src/index.tsx:54]:
```suggestion
                title="Paste"
                icon={Icon.TextDocument}
```
Would make more sense to have it only paste with this action (rewrite logic of copyAndPastePassword)

File: extensions/keepassxc/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                onAction={() => {
+                  copyAndPastePassword(entry).then(() => closeMainWindow());
+                }}
+              />
+              <ActionPanel.Item
+                title="Copy Password"
```
Comment (pernielsentikaer) [inline extensions/keepassxc/src/index.tsx:60]:
```suggestion
                title="Copy Password"
                icon={Icon.Clipboard}
```

File: extensions/keepassxc/src/index.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                shortcut={{ modifiers: ["cmd"], key: "enter" }}
+                onAction={() => {
+                  copyPassword(entry).then(() => closeMainWindow());
+                }}
+              />
+              <ActionPanel.Item
+                title="Copy Username"
```
Comment (pernielsentikaer) [inline extensions/keepassxc/src/index.tsx:67]:
```suggestion
                title="Copy Username"
                icon={Icon.Clipboard}
```

File: extensions/keepassxc/package.json [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      "title": "Path of KeepassXC.app",
+      "description": "Path of KeepassXC.app",
+      "placeholder": "Set the Path of KeepassXC.app",
+      "default": "/Applications/KeePassXC.app"
+    },
+    {
+      "name": "database",
+      "type": "textfield",
+      "required": true,
+      "title": "Keepass Database File",
```
Comment (pernielsentikaer) [inline extensions/keepassxc/package.json:32]:
```suggestion
      "title": "Path of Keepass Database File",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zkytech 👋 

We're getting pretty close, found a few more things, and then we're ready to move on!

- Can you also provide a light icon for a dark background? Otherwise, this won't look good in different themes. You can find [more info here](https://developers.raycast.com/basics/prepare-an-extension-for-store#extension-icon).

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by zkytech** [comment]
> Hi @zkytech 👋
> 
> We're getting pretty close, found a few more things, and then we're ready to move on!
> 
> * Can you also provide a light icon for a dark background? Otherwise, this won't look good in different themes. You can find [more info here](https://developers.raycast.com/basics/prepare-an-extension-for-store#extension-icon).
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

@pernielsentikaer 
Dark icon has been replaced by a light one.

**Review by pernielsentikaer — APPROVED** [review]
Hi @zkytech 👋 

Thanks for taking your time improving the UX ❤️ I think we're ready to proceed to the final step with the last review before shipping it 🚢

**Comment by fe9lix** [comment]
Hey @zkytech – would you mind putting all supporting media files in a top-level "media" folder in your extension? (runtime assets should stay in assets since they will be bundled into the extension)

**Comment by zkytech** [comment]
> Hey @zkytech – would you mind putting all supporting media files in a top-level "media" folder in your extension? (runtime assets should stay in assets since they will be bundled into the extension)

@fe9lix 

Done.

**Comment by fe9lix** [comment]
@zkytech and @pernielsentikaer Fy, I've fixed/update the title and description on master: https://github.com/raycast/extensions/commit/a69d15c6b57430234fb8cacc383502118934969e

The spelling of KeePassXC seemed to be wrong and descriptions should not be title-cased.

**Comment by rickysullivan-gallagher** [comment]
It would be nice to be able to supply a **--key-file** param.

**Pull request merged**